Userland/Utilities/unzip.cpp

```diff
@@ -162,6 +178,20 @@ int main(int argc, char** argv)
     args_parser.add_positional_argument(path, "File to unzip", "path", Core::ArgsParser::Required::Yes);
     args_parser.parse(argc, argv);
 
+    auto cwd = realpath(".", nullptr);
+    if (unveil(cwd, "rwc") < 0) {
+        fprintf(stderr, "CWD unveil\n");
+        perror("unveil");
+        return 1;
+    }
+
+    auto zip_path = realpath(path, nullptr);
+    if (unveil(zip_path, "r") < 0) {
+        fprintf(stderr, "PATH unveil\n");
+        perror("unveil");
+        return 1;
+    }
+
```

Comment:
The `unveil` state should be locked once all permitted paths have been set. Currently it is set but not locked.

```cpp
unveil(nullptr, nullptr);
```

Reply:
Good call :+1: